### PR TITLE
quic: add pending state for streams

### PR DESCRIPTION
**Pull request by addaleax**
This resolves the situation in which `openStream()` is called before
the underlying native `QuicSession` handle exists.

Fixes: https://github.com/nodejs/quic/issues/181

##### Checklist

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

File: test/parallel/test-quic-openstream-pending.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  const req = client.connect({
+    address: 'localhost',
+    port: server.address.port
+  });
+
+  const stream = req.openStream({ halfOpen: true });
+  stream.end('Hello!');
+
+  assert.strictEqual(stream.pending, true);
+  stream.on('ready', common.mustCall(() => {
```
Comment (jasnell) [inline test/parallel/test-quic-openstream-pending.js]: In the `ready` event, I'm wondering if the stream ID shouldn't be passed off to the callback, just as a helpful bit of context.

Comment (addaleax) [reply test/parallel/test-quic-openstream-pending.js]: I’m not sure if that really provides more context – it is available as `stream.id` (or `this.id`) in the event handler – but I have added a note about it in the documentation.

Comment (trivikr) [reply test/parallel/test-quic-openstream-pending.js]: Resolved in https://github.com/nodejs/quic/pull/187/commits/340eaceaebf374dcdd2cf41f920974c30cdb85d1

**Comment by addaleax** [comment]
Landed in 0e863697940c41e92750220c78049b67c821e68c

**Pull request closed without merging**